signer/src/transaction_signer.rs

```diff
@@ -594,13 +606,38 @@ where
                 self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
                     .await?;
             }
-            wsts::net::Message::DkgPublicShares(_)
-            | wsts::net::Message::DkgPrivateBegin(_)
-            | wsts::net::Message::DkgPrivateShares(_) => {
+            wsts::net::Message::DkgPrivateBegin(_) => {
+                if !chain_tip_report.sender_is_coordinator {
+                    return Err(Error::NotChainTipCoordinator);
+                }
+
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
+            wsts::net::Message::DkgPublicShares(dkg_public_shares) => {
+                let public_keys = &self.wsts_state_machines[&msg.txid].public_keys;
```

Comment:
I think that we should move away from using the index-style access, since we can panic. In this case we panic if the sender uses a transaction ID that we don't know about (purposefully or otherwise). But more generally we shy away from it and use the non-panic APIs instead.

Reply:
Done